### ISSUE TITLE
VSCode LSP request failed

### ISSUE THREAD

**Issue by morigs** [a-h/templ#956]
**Describe the bug**
 When opening minimal templ project in VSCode I get the following error: `Request textDocument/prepareCallHierarchy failed`

**To Reproduce**

```go
// main.go
package main

import (
	"context"
	"os"
)

func main() {
	component := hello("john")
	component.Render(context.Background(), os.Stdout)
}
```

```templ
package main

templ hello(name string) {
	<div>{ name }</div>
}
```

**Expected behavior**
No error

**Screenshots**
<img width="533" alt="image" src="https://github.com/user-attachments/assets/128b5af6-deab-483f-975f-e6d4c7d2c02d">

**Logs**

```
[Error - 15:12:51] Request textDocument/prepareCallHierarchy failed.
  Message: no package metadata for file file:///~/templ-tests/hello.templ
  Code: 0
```

I replace path to file from absolute to relative.

**`templ info` output**

```
(✓) os [ goos=darwin goarch=arm64 ]
(✓) go [ location=/opt/homebrew/bin/go version=go version go1.23.2 darwin/arm64 ]
(✓) gopls [ location=/Users/morigs/go/bin/gopls version=golang.org/x/tools/gopls v0.16.2 ]
(✓) templ [ location=/Users/morigs/go/bin/templ version=v0.2.778 ]
```

**Desktop (please complete the following information):**
 - OS: MacOS
 - templ CLI version: v0.2.778
- Go version `go version go1.23.2 darwin/arm64`
- `gopls` version (`golang.org/x/tools/gopls v0.16.2`)

**Additional context**

I've tried to restart vscode, close other apps, killing all gopls and templ instances, but no luck

**Comment by a-h** [comment]
Did you create a `go.mod` file by running `go mod init github.com/morigs/templ-tests`? Without a `go.mod` file, the Go language server won't be able to run.

If you can create an example and share it, I'd be able to look at it.

**Comment by morigs** [comment]
Yes, I have the `go.mod`. The problem is still present on the latest version of VSCode and Templ extension.
Here's the reproduction:
https://github.com/morigs/templ-tests

**Comment by lllama** [comment]
I'm seeing the same thing - my VS Code workspace is a level higher than my go.mod file. Don't know if that could be related.

**Comment by a-h** [comment]
I've been unable to reproduce this. I cloned the repo, and ran `go get -u github.com/a-h/templ` to update the templ runtime in the project to match the version of templ I've got in my LSP.

```
go get -u github.com/a-h/templ
go: downloading github.com/a-h/templ v0.3.833
go: upgraded github.com/a-h/templ v0.2.778 => v0.3.833
```

```
(✓) os [ goos=darwin goarch=arm64 ]
(✓) go [ location=/run/current-system/sw/bin/go version=go version go1.23.4 darwin/arm64 ]
(✓) gopls [ location=/Users/adrian/go/bin/gopls version=golang.org/x/tools/gopls v0.17.1 ]
(✓) templ [ location=/Users/adrian/go/bin/templ version=v0.3.833 ]
```

No error messages for me, everything works fine. It might only show up if a particular set of features is enabled etc.

This issue references v0.2.778. https://github.com/a-h/templ/releases/tag/v0.2.793 added workspace folder initialization (see #912), which might have had an effect, but not sure.

**Comment by a-h** [comment]
Not sure if this is relevant...

https://gophers.slack.com/archives/C2B4L99RS/p1739463683666309?thread_ts=1739435803.589139&cid=C2B4L99RS

<img width="555" alt="Image" src="https://github.com/user-attachments/assets/3983cc7e-625a-4938-892a-19479a387501" />

**Comment by morigs** [comment]
Sorry, don't use Templ now. I think we can close this as stale
